doc/integration.md

```diff
@@ -103,10 +104,87 @@ Building upon the code sample from the previous section, consider the following
 
+##Performing purchases
+The aim of this SDK is to allow users to make purchases using your application, backed by Klarna as a payment method. However, the SDK does not offer any direct methods for performing purchases as this will expose your application's private Klarna credentials. Instead, applications using the SDK are expected to work in concert with an application backend, which will perform the actual purchase requests.
+
+In this section, we will see how to communicate with such a backend and for that purpose we supply a sample backend that you can find [here](https://github.com/klarna/sample-app-backend). Reading the sample backend's documentation will allow you to fully grasp how an application using this SDK is expected to perform purchases, and you are encouraged to take a look if things become too unclear.
+
+###Signing requests
+While you can, and almost certainly will, communicate with your application's backend in a way that is different from the very simplistic approach we present here, one thing you will always have to do is sign your purchase requests. This will significantly increase your user's security while buying and the SDK makes this task incredibly easy.
+
+Let us say a user wants to make a purchase for a total of 40.50 Euros. All that's necessary to generate the relevant signature is to include the following header in your code:
+
+```objective-c
+#import "KODOriginProof.h"
+```
+
+which will allow you to perform the method call below:
+
+```objective-c
+NSString *originProof = [KODOriginProof generateWithAmount:4050 currency:@"EUR" userToken:storedToken];
+```
+
+Assume `storedToken` contains the user's token as received during registration. Note that the method expects the purchase amount to be supplied in cents. You can find the method's full documentation [here](http://cocoadocs.org/docsets/Klarna-on-Demand/0.1.2/Classes/KODOriginProof.html#//api/name/generateWithAmount:currency:userToken:).
+
+###Purchase example
+We now know how to generate the signature required for a purchase to go through. Let us see how to send it, along with other required information, to the sample backend.
+
+You will most likely have a "buy" button somewhere in your application. The code below shows how such a button might be implemented in your application's controller:
+
+```objective-c
+- (IBAction)onBuyPressed:(id)sender {
+  // create an origin proof, as seen in the previous section
+  NSString *originProof = [KODOriginProof generateWithAmount:9900 currency:@"SEK" userToken:storedToken];
+
+  // send the purchase request to the backend
+  AFHTTPRequestOperationManager *manager = [AFHTTPRequestOperationManager manager];
+  manager.requestSerializer = [AFJSONRequestSerializer serializer];
+
+  NSDictionary *params = @{
+    @"origin_proof" : originProof,
+    @"reference" :    @"TCKT0001",
```

Comment:
There should be an explanation for the `reference` parameter. Is it optional? why should the `reference` be used?